### ISSUE TITLE
Failed to send/receive M-Bus request

### ISSUE THREAD

**Issue by wmmihaa** [Apollon77/node-mbus#56]
Hi (long time no see)
We have been using ```node-mbus``` for quite a while without any issues, but I just ran into one that I can't resolve...

Our gateway which is running the node-mbus package is connected to an mbus master (Elvaco) which in turn has four of meters connected to it. We can read two of them, but while reading the others we get this:

```bash
mbus_serial_recv_frame: Attempt to read 1 bytes [len = 0]
mbus_serial_recv_frame: Got 0 byte [remaining 1, len 0]
mbus_serial_send_frame: Dumping M-Bus frame [5 bytes]: 10 7B FD 78 16
mbus_serial_recv_frame: Attempt to read 1 bytes [len = 0]
mbus_serial_recv_frame: Got 0 byte [remaining 1, len 0]
mbus_sendrecv_request: No M-Bus response frame received.
mbus_serial_send_frame: Dumping M-Bus frame [5 bytes]: 10 7B FD 78 16
mbus_serial_recv_frame: Attempt to read 1 bytes [len = 0]
mbus_serial_recv_frame: Got 1 byte [remaining 1, len 0]
68
mbus_serial_recv_frame: Attempt to read 2 bytes [len = 1]
mbus_serial_recv_frame: Got 2 byte [remaining 2, len 1]
77 77
mbus_serial_recv_frame: Attempt to read 122 bytes [len = 3]
mbus_serial_recv_frame: Got 59 byte [remaining 122, len 3]
68 08 19 72 25 70 90 68 2D 2C 1D 16 C9 00 00 00 04 13 4F 7F 1D 00 04 93 3C 00 00 00 00 04 22 53 3C 00 00 02 3B B6 00 01 67 18 22 3B 00 00 12 3B BF 04 21 67 18 11 67 1B 01 E7 FF
mbus_serial_recv_frame: Attempt to read 63 bytes [len = 62]
mbus_serial_recv_frame: Got 0 byte [remaining 63, len 62]
mbus_serial_recv_frame: Attempt to read 63 bytes [len = 62]
mbus_serial_recv_frame: Got 0 byte [remaining 63, len 62]
mbus_serial_recv_frame: Attempt to read 63 bytes [len = 62]
mbus_serial_recv_frame: Got 0 byte [remaining 63, len 62]
mbus_serial_recv_frame: Timeout
mbus_sendrecv_request: Received invalid M-Bus response frame.
mbus_serial_recv_frame: Attempt to read 1 bytes [len = 0]
mbus_serial_recv_frame: Got 0 byte [remaining 1, len 0]
mbus_serial_send_frame: Dumping M-Bus frame [5 bytes]: 10 7B FD 78 16
mbus_serial_recv_frame: Attempt to read 1 bytes [len = 0]
mbus_serial_recv_frame: Got 0 byte [remaining 1, len 0]
mbus_sendrecv_request: No M-Bus response frame received.
mbus_serial_send_frame: Dumping M-Bus frame [5 bytes]: 10 7B FD 78 16
mbus_serial_recv_frame: Attempt to read 1 bytes [len = 0]
mbus_serial_recv_frame: Got 1 byte [remaining 1, len 0]
68
mbus_serial_recv_frame: Attempt to read 2 bytes [len = 1]
mbus_serial_recv_frame: Got 2 byte [remaining 2, len 1]
77 77
mbus_serial_recv_frame: Attempt to read 122 bytes [len = 3]
mbus_serial_recv_frame: Got 59 byte [remaining 122, len 3]
68 08 19 72 25 70 90 68 2D 2C 1D 16 CA 00 00 00 04 13 4F 7F 1D 00 04 93 3C 00 00 00 00 04 22 53 3C 00 00 02 3B B6 00 01 67 18 22 3B 00 00 12 3B BF 04 21 67 18 11 67 1B 01 E7 FF
mbus_serial_recv_frame: Attempt to read 63 bytes [len = 62]
mbus_serial_recv_frame: Got 0 byte [remaining 63, len 62]
mbus_serial_recv_frame: Attempt to read 63 bytes [len = 62]
mbus_serial_recv_frame: Got 0 byte [remaining 63, len 62]
mbus_serial_recv_frame: Attempt to read 63 bytes [len = 62]
mbus_serial_recv_frame: Got 0 byte [remaining 63, len 62]
mbus_serial_recv_frame: Timeout
mbus_sendrecv_request: Received invalid M-Bus response frame.
mbus_serial_recv_frame: Attempt to read 1 bytes [len = 0]
mbus_serial_recv_frame: Got 0 byte [remaining 1, len 0]
mSB: DEBUG: [mBusSerialReader]=>Error reading sensor (689070252D2C1D16: Error: Error: Failed to send/receive M-Bus request frame[689070252D2C1D16].)

```

**Comment by Apollon77** [comment]
Can you please retry with 1.2.0? And then? maybe you can try to use some other mbus software? Is it working there? Whats the difference in send/received messages=

**Comment by wmmihaa** [comment]
Sorry I missed that you had answered... But the issue was resolved after updating to 1.2.

My apologies!
